pages/api/admin/CheckIn.js

```diff
@@ -0,0 +1,161 @@
+import { validateFunc } from "../validate";
+import { google } from "googleapis";
+
+import service from "../../../service-account.enc";
+import decrypt from "../../../utils/decrypt.js";
+
+const test = process.env.NEXT_PUBLIC_VERCEL_ENV == undefined;
+const checkin_sheet = test
+  ? process.env.SPREADSHEET_ID_TEST
+  : process.env.PANTRY_DATA_SPREADSHEET_ID;
+
+/*
+ * /api/admin/checkin
+ * req.body = { string calID }
+ */
+
+function requireParams(body, res) {
+  // makes sure that the input is in the right format
+  // returns false and an error if not a good input
+  if (body.calID) return true;
+  return false;
+}
+
+//returns the start of the week given a day
+function determineStartOfWeek(currDay) {
+  let startOfWeek = new Date(currDay.toLocaleDateString());
+  const millisecondsPerDay = 86400000;
+  const milSecondsUntilStart = millisecondsPerDay * currDay.getDay();
+  startOfWeek.setTime(startOfWeek.getTime() - milSecondsUntilStart);
+  return startOfWeek.getTime();
+}
+
+//converts from 2022-07-23T20:35:41.935Z to 7/23/2022 12:15:52
+function formatTime(timeToConvert) {
+  const formattedHours =
+    timeToConvert.getHours() +
+    ":" +
+    timeToConvert.getMinutes() +
+    ":" +
+    timeToConvert.getSeconds();
+  const formattedTime =
+    timeToConvert.toLocaleDateString() + " " + formattedHours;
+  return formattedTime;
+}
+
+//get the number of rows for the Check Out sheet
+function getNumRowsForCheckIn(properties) {
+  for (var sheet of properties["data"]["sheets"]) {
+    if (sheet.properties.title == "Check Out") {
+      return sheet.properties.gridProperties.rowCount;
+    }
+  }
+}
+
+//scans table of values for a visitor visiting more than once a week
+//loops through each pair in the table backwords until either hits calId or a week has passed
+function scanTableForVisitInPastWeek(values, startOfWeek, calId) {
+  var visitedTimes = [];
+  if (parseInt(calId) == 1) {
+    return visitedTimes;
+  }
+  for (var i = values.length - 1; i >= 0; i--) {
+    var currDate = new Date(values[i][0]);
+    if (values[i][1] == calId) {
+      if (currDate.getTime() - startOfWeek > 0) {
+        visitedTimes.push(formatTime(currDate));
```

Comment:
maybe we can use a slightly easier to read format for the return value? could drop the seconds and maybe add day of week, AM/PM?

i.e. `Mon 7/25/22 5:30 PM`